### PR TITLE
fix: run "before:spec" plugin event when experimentalInteractiveRunEvents is enabled

### LINKED ISSUE #22360: before:spec broken?
### Current behavior

When trying to run a "before:spec" function, nothing happens.

### Desired behavior

_No response_

### Test code to reproduce

setupNodeEvents(on, config) {
      on("before:spec", (spec) => {
        console.log("Running before spec: ", { spec });
});
}

(//experimentalInteractiveRunEvents is set to true.)

the log will return empty values for the spec object.

Also logged what's returned by before:run and the spec shows as an empty array.

### Cypress Version

10.1.0

### Other

_No response_

### PR DESCRIPTION
**Pull request by emilyrohrbough**
- Closes #22360

### User facing changelog

- Fix regression introduced by Cypress v10 where the `before:spec` plugin event was not triggered in open mode when `experimentalInteractiveRunEvents` was enabled. Fixed [#22360](https://github.com/cypress-io/cypress/issues/22360).

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/app/src/runner/index.ts [hunk 1]
```diff
@@ -427,12 +405,22 @@ async function executeSpec (spec: SpecFile, isRerun: boolean = false) {
...
+  // TODO: UNIFY-1318 - figure out how to manage window.config.
+  const config = getRunnerConfigFromWindow()
+
+  // this is how the Cypress driver knows which spec to run.
+  config.spec = setSpecForDriver(spec)
+
+  // creates a new instance of the Cypress driver for this spec,
+  // initializes a bunch of listeners watches spec file for changes.
+  getEventManager().setup(config)
+
```
Comment (lmiller1990) [inline packages/app/src/runner/index.ts:417]: Thanks for reducing this duplication.

File: system-tests/projects/plugin-run-events/cypress.config.js [hunk 2]
```diff
@@ -26,8 +35,14 @@ module.exports = {
         })
       })
 
-      on('before:spec', (spec) => {
-        console.log('before:spec:', spec.relative)
+      on('before:spec', (spec = {}) => {
```
Comment (lmiller1990) [inline system-tests/projects/plugin-run-events/cypress.config.js:38]:
```suggestion
      on('before:spec', (spec) => {
```

Spec should never be null, it even has the correct type definitions (yay!), so this is not necessary.

**Review by lmiller1990 — APPROVED** [review]
Tested locally, works great, thanks!

File: packages/app/src/runner/index.ts (hunk 1 above)
Comment (emilyrohrbough) [reply packages/app/src/runner/index.ts:417]: No probs! Most of this could be reduced actually but kept it simple to stay in scope

**Review by rachelruderman — APPROVED** [review]
Works great, nice to see this one go full circle from fire watch triage to fixed 😍 

Btw I noticed [unrelated flake on your CircleCI build](https://app.circleci.com/pipelines/github/cypress-io/cypress/42750/workflows/72681c80-7dcd-4c06-9d65-977835769ec4/jobs/1778748), possibly due to https://github.com/cypress-io/cypress/issues/21300. I've updated that issue with the link to your flake to help investigate it

**Pull request merged**